doctest/build.gradle

```diff
@@ -133,6 +139,49 @@ testClusters {
             }
         }))
         */
+        plugin(provider(new Callable<RegularFile>(){
+            @Override
+            RegularFile call() throws Exception {
+                return new RegularFile() {
+                    @Override
+                    File getAsFile() {
+                        // Use absolute paths
+                        String basePath = new File('.').getCanonicalPath()
+                        File dir = new File(basePath + File.separator + 'doctest' + File.separator + jsPlugin)
+
+                        // Log the directory path for debugging
+                        println("Creating directory: " + dir.getAbsolutePath())
+
+                        // Create directory if it doesn't exist
+                        if (!dir.exists()) {
+                            if (!dir.mkdirs()) {
+                                throw new IOException("Failed to create directory: " + dir.getAbsolutePath())
+                            }
+                        }
+
+                        // Define the file path
+                        File f = new File(dir, jsPlugin + '-' + opensearch_build + '.zip')
+
+                        // Download file if it doesn't exist
+                        if (!f.exists()) {
+                            println("Downloading file from: " + bwcOpenSearchJSDownload)
+                            println("Saving to file: " + f.getAbsolutePath())
+
+                            new URL(bwcOpenSearchJSDownload).withInputStream { ins ->
+                                f.withOutputStream { it << ins }
+                            }
+                        }
+
+                        // Check if the file was created successfully
+                        if (!f.exists()) {
+                            throw new FileNotFoundException("File was not created: " + f.getAbsolutePath())
+                        }
+
+                        return fileTree(f.getParent()).matching { include f.getName() }.singleFile
+                    }
+                }
+            }
+        }))
```

Comment:
What is this for? I think we want to name it by extracting or add comment to make it readable.